From 1d78d279d7c40b03247e150d005907b28ea5d1ab Mon Sep 17 00:00:00 2001
From: Shengjiu Wang <shengjiu.wang@nxp.com>
Date: Wed, 29 Aug 2018 09:57:23 +0800
Subject: [PATCH 4472/5242] MLK-19328-1: hdp: fix no sound output in second
 time

commit  bea6d0184badf1685854ba3491778c198b24a947 from
https://source.codeaurora.org/external/imx/linux-imx.git

Fixes commit 6b97462b6407 ("MLK-18690-3: hdp api: Remove
info frame API function")

The info frame API is removed, so the maximum of state->tmp
is 8.

Signed-off-by: Shengjiu Wang <shengjiu.wang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/mxc/hdp/API_Audio.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/mxc/hdp/API_Audio.c b/drivers/mxc/hdp/API_Audio.c
index b49b076..f70da60 100644
--- a/drivers/mxc/hdp/API_Audio.c
+++ b/drivers/mxc/hdp/API_Audio.c
@@ -333,7 +333,7 @@ CDN_API_STATUS CDN_API_AudioAutoConfig(state_struct *state,
 	default:
 		return ret;
 	}
-	if (state->tmp == 9) {
+	if (state->tmp == 8) {
 		state->tmp = 0;
 		return CDN_OK;
 	}
-- 
1.7.9.5

